### PR TITLE
RSDK-9470: gps module logs a lot after cord yank

### PR DESCRIPTION
**Pull request by martha-johnston**
limit error log in gps background worker to 5 logs, also change it to debug

### CODE REVIEW

**Comment by martha-johnston** [comment]
if you have a gps setup, can you test this again? if it looks good to you I can merge if it's easier for you to test

**Comment by martha-johnston** [comment]
also what are your thoughts on the log now being at the debug level?

File: gpsutils/serial_data_reader.go [hunk 1]
```diff
@@ -80,7 +81,10 @@ func (dr *SerialDataReader) backgroundWorker(cancelCtx context.Context) {
 
 		line, err := r.ReadString('\n')
 		if err != nil {
-			dr.logger.CErrorf(cancelCtx, "can't read gps serial %s", err)
+			if readErrCount < 5 {
+				dr.logger.CDebugf(cancelCtx, "can't read gps serial %s", err)
+			}
+			readErrCount++
 			continue // The line has bogus data; don't put it in the channel.
 		}
```
Comment (abe-winter) [inline gpsutils/serial_data_reader.go:92]: would add an else case here to reset the counter to 0 on success

File: gpsutils/serial_data_reader.go [hunk 2]
```diff
@@ -80,7 +81,10 @@ func (dr *SerialDataReader) backgroundWorker(cancelCtx context.Context) {
 
 		line, err := r.ReadString('\n')
 		if err != nil {
-			dr.logger.CErrorf(cancelCtx, "can't read gps serial %s", err)
+			if readErrCount < 5 {
+				dr.logger.CDebugf(cancelCtx, "can't read gps serial %s", err)
```
Comment (abe-winter) [inline gpsutils/serial_data_reader.go:86]: from ticket I assumed this would be error at first, then debug after 5

Comment (martha-johnston) [reply gpsutils/serial_data_reader.go:86]: added!

**Review by abe-winter — APPROVED** [review]
LG

want me to test?

**Comment by martha-johnston** [comment]
yes please that would be great

**Pull request merged**